### PR TITLE
Update to minecraft 1.19.2

### PR DESCRIPTION
**Pull request by hex-agon**
This is an initial attempt to update the mod to 1.19.2. I'm opening this PR as a draft as the update is currently incomplete, it's missing the biome/worldgen related changes.

This is my first time ever touching a minecraft mod so let me know if i did anything wrong.

### CODE REVIEW

**Comment by mrh0** [comment]
target branch mrh0:1.19.2

File: src/main/java/github/mrh0/beekeeping/datagen/ItemModelGenerator.java [hunk 1]
```diff
@@ -40,19 +36,20 @@ protected void registerModels() {
     }
 
     private ItemModelBuilder simpleItem(Item item) {
-        return withExistingParent(item.getRegistryName().getPath(),
+        return withExistingParent(item.toString(),
```
Comment (mrh0) [inline src/main/java/github/mrh0/beekeeping/datagen/ItemModelGenerator.java]: Not sure this is correct

Comment (mrh0) [reply src/main/java/github/mrh0/beekeeping/datagen/ItemModelGenerator.java]: Same with the stuff below

**Comment by mrh0** [comment]
Also, currently working on some new stuff for the mod in the 1.18.2 branch

File: src/main/java/github/mrh0/beekeeping/datagen/ItemModelGenerator.java (hunk 1 above)
Comment (hex-agon) [reply src/main/java/github/mrh0/beekeeping/datagen/ItemModelGenerator.java]: I believe they would work as the implementation of the `item.toString` is:
```
   public String toString() {
      return Registry.ITEM.getKey(this).getPath();
   }
```

Though this is possibly a little fragile and prone to breaking in future changes, i've replaced it with forge registry calls now.

**Comment by hex-agon** [comment]
Just commenting on what's my current status, i've been looking at this sparingly and the current issues are:
* Biomes went through a big revamp in 1.19 on how they're actually structured, there's no enumeration (as far as i've found) on which vanilla biomes exist, everything is now data driven.

My current idea is to move the beehive generation logic into biome modifier and use biome tags to do the placement picking, i think this should work...

**Comment by hex-agon** [comment]
I got to a point where i can launch the game and beehives do generate but i think there's still room for improvement:

**Review by mrh0 — APPROVED** [review]

**Pull request merged**